### PR TITLE
LPD-27768 Taken into account ctCollectionId during DDMFormInstanceUpgradeProcess v4_3_1 of the DDMFormInstance and DDMFormInstanceVersion entities

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPD-27768

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceUpgradeProcessTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		ddmFormValues.addDDMFormFieldValue(
+			DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+				"workflowDefinition",
+				StringBundler.concat(
+					"[\"", RandomTestUtil.randomString(), "@",
+					RandomTestUtil.randomInt(), "\"]")));
+
+		_ddmFormInstance = DDMFormInstanceTestUtil.addDDMFormInstance(
+			ddmForm, _group, ddmFormValues, TestPropsValues.getUserId());
+
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceUpgradeProcessTest.java:76]: @p-albuquerque I noticed that you are defining the `ddmFormValues` for the `ddmFormInstance` **settings**, so it makes more sense to do something like:

```
DDMForm ddmForm = DDMFormFactory.create(DDMFormInstanceSettings.class);

DDMFormValues ddmFormValues = DDMFormValuesTestUtil.createDDMFormValues(ddmForm);

ddmFormValues.addDDMFormFieldValue(
	DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
		"workflowDefinition", ....);

_ddmFormInstance = DDMFormInstanceTestUtil.addDDMFormInstance(
	DDMFormTestUtil.createDDMForm(RandomTestUtil.randomString()), 
	_group, ddmFormValues, TestPropsValues.getUserId());
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceUpgradeProcessTest.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	@Override
+	protected CTModel<?> updateCTModel(CTModel<?> ctModel) throws Exception {
+		DDMFormInstance ddmFormInstance = (DDMFormInstance)ctModel;
+
+		return DDMFormInstanceTestUtil.updateDDMFormInstance(
+			ddmFormInstance.getFormInstanceId(),
+			DDMFormValuesTestUtil.createDDMFormValuesWithRandomValues(
+				DDMFormTestUtil.createDDMForm(RandomTestUtil.randomString())));
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceUpgradeProcessTest.java:100]: @p-albuquerque can we change it to `DDMFormFactory.create(DDMFormInstanceSettings.class)`?

Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceUpgradeProcessTest.java:100]: or even use the `DDMFormInstanceTestUtil.createSettingsDDMFormValues()`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceVersionFormInstanceUpgradeProcessTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class DDMFormInstanceVersionFormInstanceUpgradeProcessTest
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceVersionFormInstanceUpgradeProcessTest.java:32]: @p-albuquerque please rename it to `DDMFormInstanceVersionDDMFormInstanceUpgradeProcessTest`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceVersionFormInstanceUpgradeProcessTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	@Override
+	protected CTModel<?> updateCTModel(CTModel<?> ctModel) throws Exception {
+		DDMFormInstanceVersion ddmFormInstanceVersion =
+			(DDMFormInstanceVersion)ctModel;
+
+		ddmFormInstanceVersion.setSettings(
+			DDMFormValuesSerializeUtil.serialize(
+				DDMFormValuesTestUtil.createDDMFormValuesWithRandomValues(
+					DDMFormTestUtil.createDDMForm(
+						RandomTestUtil.randomString())),
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceVersionFormInstanceUpgradeProcessTest.java:64]: @p-albuquerque same as above

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceUpgradeProcessTest.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceUpgradeProcessTest.java:76]: @carolmariaabb Do you mean:

```java
DDMFormValues ddmFormValues = DDMFormValuesTestUtil.createDDMFormValues(
	DDMFormFactory.create(DDMFormInstanceSettings.class));

ddmFormValues.addDDMFormFieldValue(
	DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
		"workflowDefinition",
		...));

_ddmFormInstance = DDMFormInstanceTestUtil.addDDMFormInstance(
	DDMFormTestUtil.createDDMForm(RandomTestUtil.randomString()),
	_group, ddmFormValues, TestPropsValues.getUserId());
```

Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v4_3_1/test/DDMFormInstanceUpgradeProcessTest.java:76]: @p-albuquerque yes

**Comment by p-albuquerque** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**